includes/class.imageuploader.php

```diff
@@ -95,22 +102,18 @@ public function save() {
         // check if file with same name exists in upload path
         if (is_file($image_path)) {
             $this->url = $image_url;
-            curl_close($ch);
-            return true;
+            // If the image already exists, get the id to link it to the post
+            $attachment_id = attachment_url_to_postid($image_url);
```

Comment:
@gregory-claeyssens  this is the change based on our chat yesterday

Reply:
If it works, this is looking food imo :)

Reply:
Looking food ;) ⬆️ 

yes, so testing **with out** this change : 
featured image is not attached in wp in some cases ( already exists or already downloaded)

Testing **with** this change: 
Featured image seems to always be there in all my tests